docs: add smart router component (#10097)
langflow-ai/langflow: docs/docs/Components/components-logic.mdx

| dynamic inputs    | Various  | Input parameter. Additional inputs are generated based on the selected flow. |
| run_outputs  | A `List` of types (`Data`, `Message`, or `DataFrame`)  | Output parameter. All outputs are generated from running the flow.  |

## Smart Router {#smart-router}

The **Smart Router** component builds on the [**If-Else** component](#if-else), but instead of string matching, the **Smart Router** uses a connected [**Language Model** component](/components-models) to categorize and route incoming messages.

> To use the **Smart Router**, replace the **If-Else** component in a flow like the [Incoming messages flow](#use-the-if-else-component-in-a-flow). Instead of sorting by regex, use the **Routes** table to define the outputs for your messages, and connect a **Language Model** component to provide the sorting intelligence.

Reviewer: ```suggestion
You can use the **Smart Router** component anywhere you would use the **If-Else** component.
For an example, create [If-Else component example flow](#use-the-if-else-component-in-a-flow), then replace the **If-Else** component with a **Smart Router** component.
Instead of a regex, use the **Routes** table to define the outputs for your messages.
Finally, connect a **Language Model** component to provide the sorting intelligence.
```

### Smart Router parameters

<PartialParams />

| Name                | Type     | Description                                                       |
|---------------------|----------|-------------------------------------------------------------------|

> | llm                 | Language Model | Input parameter. The language model to use for categorization. Required. |

Reviewer: ```suggestion
| llm                 | [LanguageModel](/data-types#languagemodel | Input parameter. The language model to use for categorization. Required. |
```

| input_text          | String   | Input parameter. The primary text input for categorization. Required. |
| routes              | Table    | Input parameter. Table defining categories and optional output values. Each row should have a route/category name and optionally a custom output value. Required. |
| message             | Message  | Input parameter. Optional override message that replaces both the Input and Output Value for all routes when filled. Advanced. |
| enable_else_output  | Boolean  | Input parameter. Include an Else output for cases that don't match any route. Default: false. |
| custom_prompt       | String   | Input parameter. Additional instructions for LLM-based categorization. Use `{input_text}` for the input text and `{routes}` for the available categories. |

> | default_result      | Message  | Output parameter. The Else output. Only available when `enable_else_output` is `true`. |

Reviewer: ```suggestion
| default_result      | Message  | Output parameter. The Else output. Only available when `enable_else_output` is `true`. Otherwise, output is produced and routed according to the `routes` parameter. |
```
